pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -138,3 +175,80 @@ func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconci
 	// Don't requeue if secret is created successfully
 	return reconcile.Result{}, err
 }
+
+// getRoute returns an OpenShift route object. It waits until the .spec.host value exists to avoid possible race conditions, fails otherwise.
+func (r *ReconcileSecretGenerator) getRoute(name string, namespace string) (*routev1.Route, error) {
+	oauthClientRoute := &routev1.Route{}
+	// Get spec.host from route
+	err := wait.PollImmediate(resourceRetryInterval, resourceRetryTimeout, func() (done bool, err error) {
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: name,
+			Namespace: namespace}, oauthClientRoute)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		} else if oauthClientRoute.Spec.Host == "" {
+			return false, nil
+		} else {
+			return true, nil
+		}
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return oauthClientRoute, err
+}
+
+func (r *ReconcileSecretGenerator) createOAuthClient(secretName string, name string, secret string, uri string) error {
+	// Create OauthClient resource
```

Comment:
There are multiple places with this typo, `OauthClient` should be `OAuthClient`.